optuna/integration/tensorboard.py

```diff
@@ -0,0 +1,155 @@
+import optuna
+from optuna import type_checking
+import os
+
+if type_checking.TYPE_CHECKING:
+    from typing import Any  # NOQA
+    from typing import Dict  # NOQA
+    from typing import Optional  # NOQA
+    from typing import Tuple  # NOQA
+
+try:
+    import tensorboard
+    from tensorboard.plugins.hparams import api as hp
+    import tensorflow as tf
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    _available = False
+    hp = object
+    tensorboard = object
+    tensorflow = object
+
+
+def _check_tensorboard_availability() -> None:
+
+    if not _available:
+        raise ImportError(
+            "TensorBoard is not available. Please install TensorBoard to use this "
+            "feature. It can be installed by executing `$ pip install "
+            "tensorboard`. For further information, please refer to the installation guide "
+            "of TensorBoard. (The actual import error is as follows: " + str(_import_error) + ")"
+        )
```

Comment:
We can use DRY import. This feature is recently introduced in PR #1315.  

```suggestion
from optuna._imports import try_import

with try_import() as _imports:
    from tensorboard.plugins.hparams import api as hp
    import tensorflow as tf
```